### PR TITLE
Replace `newTranslatableError` with `UserFriendlyError` (#10440

### PR DESCRIPTION
**Pull request by MadLittleMods**
Swapping out [`newTranslatableError('foo')`](https://github.com/matrix-org/matrix-react-sdk/blob/5c47142dab202a48aec6ef1369c7e0ab079107e4/src/languageHandler.tsx#L52-L63) where we lose error types with `new UserFriendlyError("This email address is already in use" , { cause: err });` where we can still get translations and maintain the underlying cause info and type semantics.

*Spawning from https://github.com/matrix-org/matrix-react-sdk/pull/10432#discussion_r1145538597 which has more usage if you're looking for something that actually uses the `cause` and type niceness*

## Todo

 - [x] Make sure `i18n` includes strings from `UserFriendlyError`
    - Requires https://github.com/matrix-org/matrix-web-i18n/pull/6

## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [ ] Linter and other CI checks pass
-   [ ] ~~Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))~~

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: src/components/views/right_panel/UserInfo.tsx [hunk 1]
```diff
@@ -448,7 +448,15 @@ export const UserOptionsSection: React.FC<{
...
-                            throw new Error(inviter.getErrorText(member.userId) ?? undefined);
+                            const errorStringFromInviterUtility = inviter.getErrorText(member.userId);
+                            if (errorStringFromInviterUtility) {
+                                throw new Error(errorStringFromInviterUtility);
+                            } else {
+                                throw new UserFriendlyError(
+                                    `User (%(user)s) did not end up as invited to %(roomId)s but no error was given from the inviter utility`,
+                                    { user: member.userId, roomId, cause: undefined },
+                                );
+                            }
```
Comment (MadLittleMods) [inline src/components/views/right_panel/UserInfo.tsx:459]: The [remaining TS strict errors](https://github.com/matrix-org/matrix-react-sdk/actions/runs/4535420375/jobs/7990846522?pr=10440) are unrelated and tracked by https://github.com/vector-im/element-web/issues/24899

And there are some nits from files we didn't even change

File: src/SlashCommands.tsx [hunk 2]
```diff
@@ -30,7 +30,7 @@ import { SlashCommand as SlashCommandEvent } from "@matrix-org/analytics-events/
 
 import { MatrixClientPeg } from "./MatrixClientPeg";
 import dis from "./dispatcher/dispatcher";
-import { _t, _td, ITranslatableError, newTranslatableError } from "./languageHandler";
+import { _t, _td, UserFriendlyError } from "./languageHandler";
```
Comment (MadLittleMods) [inline src/SlashCommands.tsx:33]: SonarCloud coverage complaint and most is from this file, https://github.com/matrix-org/matrix-react-sdk/pull/10440/checks?check_run_id=12312160706

Adding test coverage here is pretty tangent to the this PR so I'm going to leave it as-is.

**Review by dbkr — APPROVED** [review]

File: package.json [hunk 3]
```diff
@@ -28,7 +28,7 @@
     "matrix_lib_main": "./lib/index.ts",
     "matrix_lib_typings": "./lib/index.d.ts",
     "matrix_i18n_extra_translation_funcs": [
-        "newTranslatableError"
+        "UserFriendlyError"
```
Comment (MadLittleMods) [inline package.json:31]: @kerryarchibald This PR requires a `element-web-app-team` approval which you're part of. I just want to make sure this is still on your list even after @dbkr approved?

**Review by kerryarchibald — APPROVED** [review]

File: package.json (hunk 3 above)
Comment (MadLittleMods) [reply package.json:31]: Thanks for the review pass @kerryarchibald ✅

**Comment by MadLittleMods** [comment]
Thank you @dbkr and @kerryarchibald for the review and @t3chguy for the merge 🐁

Thank you @justjanne for fixing up the downstream usage in `element-web` https://github.com/vector-im/element-web/pull/25002 🧐

**Pull request merged**